src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -336,4 +430,312 @@ private void populateQSYSMonitoringMetrics(Map<String, String> stats) {
+		controlInfo.setName(controlInfo.getName().replace(QSYSCoreConstant.HASH, QSYSCoreConstant.TIDE));
+		stats.put(QSYSCoreConstant.GAIN_LABEL + controlInfo.getName() + QSYSCoreControllingMetric.CURRENT_GAIN_VALUE.getMetric(), controlInfo.getGainString());
+
+		if (canControlGain) {
+			stats.put(QSYSCoreConstant.GAIN_LABEL + controlInfo.getName() + QSYSCoreControllingMetric.GAIN_VALUE_CONTROL.getMetric(), "");
+			controllableProperties.add(createSlider(QSYSCoreConstant.GAIN_LABEL + controlInfo.getName() + QSYSCoreControllingMetric.GAIN_VALUE_CONTROL.getMetric(),
+					controlInfo.getMinGain(), controlInfo.getMaxGain(), Float.parseFloat(controlInfo.getMinGain()), Float.parseFloat(controlInfo.getMaxGain()),
+					(float) controlInfo.getGainValue()));
+		}
+
+		stats.put(QSYSCoreConstant.GAIN_LABEL + controlInfo.getName() + QSYSCoreControllingMetric.MUTE_CONTROL.getMetric(), "");
+		controllableProperties.add(
+				createSwitch(QSYSCoreConstant.GAIN_LABEL + controlInfo.getName() + QSYSCoreControllingMetric.MUTE_CONTROL.getMetric(), (int) controlInfo.getMutePosition()));
+	}
+
+	/**
+	 * This method used to populate unavailable gain controlling group
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param namedGainComponent is an unavailable gain component
+	 * @param errorMessage is an error message want to populate
+	 */
+	private void populateUnavailableGainControllingGroup(Map<String, String> stats, String namedGainComponent, String errorMessage) {
+		stats.put(QSYSCoreConstant.GAIN_LABEL + namedGainComponent + QSYSCoreControllingMetric.ERROR_MESSAGE.getMetric(), errorMessage);
+	}
+
+	/**
+	 * This method used to build a control command to send to device
+	 *
+	 * @param namedComponent is named component want to send
+	 * @param method is the control method (get and set)
+	 * @param value is the value want to set if control method is set
+	 * @param property is the property want to control (gain, mute,...)
+	 * @return ControlRequest is the request to send to device
+	 */
+	public ControlRequest buildSendControlCommand(String namedComponent, QSYSCoreControllingMethod method, Double value, String property) {
+		ControlProperty controlProperty = new ControlProperty(property);
+
+		if (value != null) {
+			controlProperty.setValue(value);
+		}
+
+		ControlParam params = new ControlParam(namedComponent);
+		params.add(controlProperty);
+		return new ControlRequest(method.getName(), params);
+	}
+
+	/**
+	 * This method is used to get responses from device
+	 *
+	 * @param request is the RPC request to be sent
+	 * @param responseClass is the response class for mapping
+	 * @return Rpc is the json rpc of response
+	 */
+	public Rpc getResponsesFromDevice(Rpc request, Class<? extends Rpc> responseClass) {
+		try {
+			ObjectMapper mapper = new ObjectMapper();
```

Comment:
you can store the ObjectMapper as local variable

Reply:
ok thank you, I will store it as local variable